src/test/java/org/jenkinsci/plugins/pipeline/milestone/MilestoneStepMigrationTest.java

```diff
@@ -0,0 +1,43 @@
+package org.jenkinsci.plugins.pipeline.milestone;
+
+import hudson.model.ParametersDefinitionProperty;
+import hudson.model.Result;
+import hudson.model.StringParameterDefinition;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.test.steps.SemaphoreStep;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runners.model.Statement;
+import org.jvnet.hudson.test.BuildWatcher;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.RestartableJenkinsRule;
+import org.jvnet.hudson.test.recipes.LocalData;
+import org.jvnet.hudson.test.recipes.WithTimeout;
+
+public class MilestoneStepMigrationTest {
+
+
+    @ClassRule
+    public static BuildWatcher buildWatcher = new BuildWatcher();
+
+    @Rule
+    public RestartableJenkinsRule story = new RestartableJenkinsRule();
+
+
+    @Test
+    //@WithTimeout(60)
+    @LocalData("migration_1_3to1_4")
+    public void migration_from_1_3to1_4() {
+        story.then(r -> {
+            WorkflowJob p = (WorkflowJob) story.j.jenkins.getItem("p");
+            WorkflowRun b3 = p.scheduleBuild2(0).waitForStart();
+            SemaphoreStep.waitForStart("wait/2", b3);
```

Comment:
@jglick 
I also want to force jenkins to restore previous build (`p#2`) which is passed the milestone and waits for a semaphore. 

But test hangs and in logs I've got this:
```
3.766 [id=16]	WARNING	o.j.p.w.f.FlowExecutionList$ItemListenerImpl$1#onFailure: Failed to load CpsFlowExecution[Owner[p/2:p #2]]
java.util.concurrent.RejectedExecutionException
	at hudson.remoting.SingleLaneExecutorService.execute(SingleLaneExecutorService.java:100)
	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:112)
	at com.google.common.util.concurrent.ForwardingExecutorService.submit(ForwardingExecutorService.java:110)
	at jenkins.util.InterceptingExecutorService.submit(InterceptingExecutorService.java:49)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution$4.onSuccess(CpsFlowExecution.java:655)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution$4.onSuccess(CpsFlowExecution.java:651)
	at org.jenkinsci.plugins.workflow.support.concurrent.Futures$1.run(Futures.java:150)
	at com.google.common.util.concurrent.MoreExecutors$SameThreadExecutorService.execute(MoreExecutors.java:253)
	at com.google.common.util.concurrent.ExecutionList$RunnableExecutorPair.execute(ExecutionList.java:149)
	at com.google.common.util.concurrent.ExecutionList.add(ExecutionList.java:105)
	at com.google.common.util.concurrent.AbstractFuture.addListener(AbstractFuture.java:155)
	at org.jenkinsci.plugins.workflow.support.concurrent.Futures.addCallback(Futures.java:160)
	at org.jenkinsci.plugins.workflow.support.concurrent.Futures.addCallback(Futures.java:90)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.runInCpsVmThread(CpsFlowExecution.java:651)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.getCurrentExecutions(CpsFlowExecution.java:721)
	at org.jenkinsci.plugins.workflow.flow.FlowExecutionList$ItemListenerImpl.onLoaded(FlowExecutionList.java:180)
	at jenkins.model.Jenkins.<init>(Jenkins.java:999)
	at hudson.model.Hudson.<init>(Hudson.java:86)
	at org.jvnet.hudson.test.JenkinsRule.newHudson(JenkinsRule.java:675)
	at org.jvnet.hudson.test.JenkinsRule.before(JenkinsRule.java:402)
	at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:595)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
   3.839 [p #3] Started
Blocking wait/1
```